src/js/src/components/SearchBox.vue

```diff
@@ -92,153 +92,65 @@ export default {
-        this.clearResults()
-        this.updateQuery(this.futureQuery)
-        this.updateSolrSettingOffset(0)
-        this.updateSolrSettingGrouping(this.futureSolrSettings.grouping)
-        this.updateSolrSettingUrlSearch(this.futureSolrSettings.urlSearch)
-        this.updateSolrSettingImgSearch(this.futureSolrSettings.imgSearch)
-        if(this.solrSettings.imgSearch) {
-           this.requestImageSearch({query:this.query})
-           this.$_pushSearchHistory('SolrWayback', this.query, this.searchAppliedFacets, this.solrSettings)
-        }
-        else if(this.solrSettings.urlSearch) {
-          this.preNormalizeQuery = this.futureQuery
-          let queryString = ''
-          if(this.futureQuery.substring(0,10) === 'url_norm:"') {
-            queryString = this.futureQuery.replace('url_norm:"', '')
-            queryString.substring(queryString.length-1, queryString.length) === '"' ? queryString = queryString.slice(0,-1) : null
-          }
-          else {
-            queryString = this.futureQuery
-          }
-          if(this.validateUrl(queryString)) {
-            this.requestUrlSearch({query:queryString, facets:this.searchAppliedFacets, options:this.solrSettings})
-            this.requestFacets({query:'url_norm:"' + queryString + '"', facets:this.searchAppliedFacets, options:this.solrSettings})
-            this.$_pushSearchHistory('SolrWayback', queryString, this.searchAppliedFacets, this.solrSettings)
-          }
-          else {
-            this.setNotification({
-          	title: 'We are so sorry!',
-            text: 'This query is not valid. the url must start with \'http://\' or \'https://\'',
-            type: 'error',
-            timeout: false
-          })
-          }
-        }
-        else {
-          this.requestSearch({query:this.futureQuery, facets:this.searchAppliedFacets, options:this.solrSettings})
-          this.requestFacets({query:this.futureQuery, facets:this.searchAppliedFacets, options:this.solrSettings})
-          this.$_pushSearchHistory('SolrWayback', this.query, this.searchAppliedFacets, this.solrSettings)
-        }
-      }
-    },
     selectSearchMethod(selected) {
       console.log(selected)
       if(selected === 'imgSearch') {
-        this.updateFutureSolrSettingImgSearch(!this.futureSolrSettings.imgSearch)
-        this.futureSolrSettings.imgSearch ? this.updateFutureSolrSettingUrlSearch(false) : null
+        this.updateSolrSettingImgSearch(!this.solrSettings.imgSearch)
+        this.solrSettings.imgSearch ? this.updateSolrSettingUrlSearch(false) : null
       }
       else if(selected === 'urlSearch') {
-        this.updateFutureSolrSettingUrlSearch(!this.futureSolrSettings.urlSearch)
-        this.futureSolrSettings.urlSearch ? this.updateFutureSolrSettingImgSearch(false) : null
+        this.updateSolrSettingUrlSearch(!this.solrSettings.urlSearch)
+        this.solrSettings.urlSearch ? this.updateSolrSettingImgSearch(false) : null
```

Comment:
True, I did it because I'm not quite sure about the heaviness of the operations. The current logic performs a check, and then IF it's necessary performs the action of changing the variable. If we remove the check (I've done so now in the PR), we always perform the action... Its very small thing, I was just curious.